row_cache.cc

```diff
@@ -47,7 +47,7 @@ using namespace cache;
 flat_mutation_reader
 row_cache::create_underlying_reader(read_context& ctx, mutation_source& src, const dht::partition_range& pr) {
     ctx.on_underlying_created();
-    return src.make_reader(_schema, pr, ctx.slice(), ctx.pc(), ctx.trace_state(), streamed_mutation::forwarding::yes);
+    return src.make_reader(_schema, pr, ctx.slice(), ctx.pc(), ctx.trace_state(), streamed_mutation::forwarding::yes, ctx.fwd_mr());
```

Comment:
This is not correct, even if ctx.fwd_mr() is false, the underlying reader needs to be fast-forwardable in range scans.

Should be `mutation_reader::forwarding(ctx.is_range_query() || ctx.fwd_mr())`

Isn't this parameter currently ignored?

See `[PATCH v1] cache, sstables: Allow fast-forwarding from a singular partition range`

Reply:
I think it is: https://github.com/scylladb/scylla/pull/5452#discussion_r356458909
I think it would still be good to propagate it.